src/main/kotlin/com/exactpro/th2/check1/rule/sequence/NoMessageCheckTask.kt

```diff
@@ -0,0 +1,134 @@
+        protoPreMessageFilter.toCompareSettings()
+    )
+
+    private val metadataPreFilter: SailfishFilter? = protoPreMessageFilter.metadataFilterOrNull()?.let {
+        SailfishFilter(
+            converter.fromMetadataFilter(it, VerificationUtil.METADATA_MESSAGE_NAME),
+            it.toComparisonSettings()
+        )
+    }
+
+    private lateinit var preFilterEvent: Event
+    private lateinit var resultEvent: Event
+
+    private var preFilterMessagesCounter: Int = 0
+    private var extraMessagesCounter: Int = 0
+
+    init {
+        rootEvent
+            .name("No message check $sessionKey")
+            .type("noMessageCheck")
+    }
+
+    override fun onStart() {
+        super.onStart()
+        preFilterEvent = Event.start()
+            .type("preFiltering")
+            .bodyData(protoPreMessageFilter.toTreeTable())
+        rootEvent.addSubEvent(preFilterEvent)
+        resultEvent = Event.start()
+            .type("noMessagesCheckResult")
+        rootEvent.addSubEvent(resultEvent)
+    }
+
+    override fun onNext(messageContainer: MessageContainer) {
+        LOGGER.debug(
+            "Received message with id: {}",
+            TextFormat.shortDebugString(messageContainer.protoMessage.metadata.id)
+        )
+        val result = matchFilter(
+            messageContainer,
+            messagePreFilter,
+            metadataPreFilter,
+            matchNames = false,
+            significant = false
+        )
+        val comparisonContainer = ComparisonContainer(messageContainer, protoPreMessageFilter, result)
+        if (comparisonContainer.fullyMatches) {
+            preFilterMessagesCounter++
+            preFilterEvent.messageID(comparisonContainer.protoActual.metadata.id)
+        } else {
+            extraMessagesCounter++
+            resultEvent.messageID(comparisonContainer.protoActual.metadata.id)
+        }
+    }
+
+    override fun completeEvent(canceled: Boolean) {
+        preFilterEvent.name("Prefilter: $preFilterMessagesCounter messages were filtered.")
+
+        if (extraMessagesCounter == 0)
+            resultEvent.status(Event.Status.PASSED).name("Check passed")
+        else
+            resultEvent.status(Event.Status.FAILED).name("Check failed: $extraMessagesCounter extra messages were found.")
+    }
+
+    private fun ProtoToIMessageConverter.fromProtoPreFilter(protoPreMessageFilter: RootMessageFilter): IMessage =
+        fromProtoFilter(protoPreMessageFilter.messageFilter, PRE_FILTER_MESSAGE_NAME)
+
+    private fun PreFilter.toRootMessageFilter() = RootMessageFilter.newBuilder()
+        .setMessageType(PRE_FILTER_MESSAGE_NAME)
+        .setMessageFilter(toMessageFilter())
+        .also {
+            if (hasMetadataFilter()) {
+                it.metadataFilter = metadataFilter
+            }
+        }
+        .build()
+
+    private fun PreFilter.toMessageFilter() = MessageFilter.newBuilder()
+        .putAllFields(fieldsMap)
+        .build()
```

Comment:
The same methods are used in CheckSequenceRule. We probably should extract those methods into a utility class or into the AbstractCheckTask and reuse them